.gitlab-ci.yml

```diff
@@ -1,259 +1,361 @@
+  paths:
+    - 'vendor/ruby'
+
+#=======================================================================
+# Anchors
+
+.lint_base: &lint_base
+  stage: 'lint'
+  tags: ['docker']
+  script:
+    - 'bundle exec rake syntax'
+    - 'bundle exec rake lint'
   cache:
-    untracked: true
-    # A broad attempt at caching between runs (ala Travis CI)
-    key: "${CI_PROJECT_NAMESPACE}__bundler"
+    policy: 'pull'
+  dependencies: []
+  artifacts:
+    when: 'always'
     paths:
-      - '.vendor'
-      - 'vendor'
-
-.setup_bundler_env: &setup_bundler_env
-  before_script:
-    - 'echo Files in cache: $(find .vendor | wc -l) || :'
-    - 'export GEM_HOME=.vendor/gem_install'
-    - 'export BUNDLE_CACHE_PATH=.vendor/bundler'
-    - 'declare GEM_BUNDLER_VER=(-v ''~> ${BUNDLER_VERSION:-1.16.0}'')'
-    - declare GEM_INSTALL=(gem install --no-document)
-    - declare BUNDLER_INSTALL=(bundle install --no-binstubs --jobs $(nproc) --path=.vendor "${FLAGS[@]}")
-    - gem list -ie "${GEM_BUNDLE_VER[@]}" --silent bundler || "${GEM_INSTALL[@]}" --local "${GEM_BUNDLE_VER[@]}" bundler || "${GEM_INSTALL[@]}" "${GEM_BUNDLE_VER[@]}" bundler
-    - 'rm -rf pkg/ || :'
-    - bundle check || rm -f Gemfile.lock && ("${BUNDLER_INSTALL[@]}" --local || "${BUNDLER_INSTALL[@]}")
-
-
-.validation_checks: &validation_checks
-  script:
-    - bundle exec rake syntax
-    - bundle exec rake check:dot_underscore
-    - bundle exec rake check:test_file
-    - bundle exec rake pkg:check_version
-    - bundle exec rake pkg:compare_latest_tag
-    - bundle exec rake lint
-    - bundle exec rake clean
-    - bundle exec puppet module build
-
-.spec_tests: &spec_tests
+      - 'Gemfile.lock'
+
+.unit_base: &unit_base
+  stage: 'unit'
+  tags: ['docker']
+  variables:
+    STRINGIFY_FACTS: 'no'
+    TRUSTED_NODE_DATA: 'yes'
   script:
-    - bundle exec rake spec
+    - 'bundle exec rake spec'
+  cache:
+    policy: 'pull'
+  dependencies: []
+  artifacts:
+    when: 'always'
+    paths:
+      - 'Gemfile.lock'
 
-stages:
-  - validation
-  - unit
-  - acceptance
-  - deploy
-
-# Puppet 4.7 for PE 2016.4 LTS Support (EOL: 2018-10-21)
-# See: https://puppet.com/misc/puppet-enterprise-lifecycle
-# --------------------------------------
-pup4_7-validation:
-  stage: validation
-  tags:
-    - docker
-  image: ruby:2.1
+.acceptance_base: &acceptance_base
+  stage: 'acceptance'
+  tags: ['beaker']
   variables:
-    PUPPET_VERSION: '~> 4.7.0'
-  <<: *cache_bundler
-  <<: *setup_bundler_env
-  <<: *validation_checks
+    BUNDLE_WITHOUT: 'development'
+  cache:
+    policy: 'pull'
+  dependencies: []
+  artifacts:
+    when: 'always'
+    paths:
+      - 'Gemfile.lock'
 
-pup4_7-unit:
-  stage: unit
-  tags:
-    - docker
-  image: ruby:2.1
+# ----------------------------------------------------------------------
+# Version Matrix
+#
+# Selections are based on the component version combinations included with
+# currently supported releases of SIMP and Puppet Enterprise.
```

Comment:
Slimming down this section from the [first draft][1] was a step in the right direction.

However, the original comment is still more effective in one simple line:

```bash
# The testing matrix considers ruby/puppet versions supported by SIMP and PE`
```
Than the new comment is with four:

```bash
# Version Matrix 
#
# Selections are based on the component version combinations included with
# currently supported releases of SIMP and Puppet Enterprise.
```
The original version was concise and clear.  The new version is wordier and ambiguous in comparison: 
* The passive voice adds words and makes the point less direct.
* The well-known acronym "PE" is unnecessarily expanded into full words.
* The generic word "Selections" replaced the specific  term "testing matrix."
* As a consequence, two additional lines (starting with "`# Version Matrix`") were needed to reinstate the information that "Selections" had removed.

[1]: https://github.com/simp/pupmod-simp-simplib/blob/03c597f504bd460989d73b0160c3935e2522487f/.gitlab-ci.yml#L86-L93